services/src/test/java/org/exoplatform/addons/gamification/storage/dao/RuleDAOTest.java

```diff
@@ -137,6 +138,12 @@ public void testFindRulesByFilter() {
     assertEquals(ruleDAO.findRulesByFilter(filter, 0, 10).size(), 2);
     newRule("rule3", "domain3", 1l);
     assertEquals(ruleDAO.findRulesByFilter(filter, 0, 10).size(), 2);
+    filter.setFilterType(FilterType.STARTED);
+    assertEquals(ruleDAO.findRulesByFilter(filter, 0, 10).size(), 2);
+    filter.setFilterType(FilterType.NOT_STARTED);
+    assertEquals(ruleDAO.findRulesByFilter(filter, 0, 10).size(), 0);
+    filter.setFilterType(FilterType.ENDED);
+    assertEquals(ruleDAO.findRulesByFilter(filter, 0, 10).size(), 0);
```

Comment:
Here we tested retrieving **0** rules for `NOT_STARTED` and `ENDED` fitler types. Can you make different test cases to cover the Test case where we can have results returned by using `NOT_STARTED` and `ENDED` filter types.

Reply:
done!